kimchi/src/circuits/lookup/tables/mod.rs

```diff
@@ -78,15 +78,19 @@ impl<F> LookupTable<F>
 where
     F: FftField,
 {
-    /// Return true if the table has an entry containing all zeros.
+    /// Return true if the table has an entry (row) containing all zeros.
     pub fn has_zero_entry(&self) -> bool {
         // reminder: a table is written as a list of columns,
         // not as a list of row entries.
         for row in 0..self.len() {
+            let mut row_zero = true;
```

Comment:
This could probably be made clearer by hiding this state in an accumulator. For example
```rust
for row in 0..self.len() {
    if self.data.iter().all(|col| col[row].is_zero()) {
        return true;
    }
}
false
```